honeybee_energy/measure.py

```diff
@@ -0,0 +1,428 @@
+        self._display_name = root.find('display_name').text
+        self._description = root.find('description').text
+        self._measure_type = None
+        for atr in root.find('attributes'):
+            if atr.find('name').text == 'Measure Type':
+                self._type = atr.find('value').text
+
+        # parse the measure arguments
+        self._arguments = []
+        for arg in root.find('arguments'):
+            arg_obj = MeasureArgument(arg)
+            if arg_obj.model_dependent:
+                # TODO: Figure out how to implement model-dependent arguments
+                raise NotImplementedError(
+                    'Model dependent arguments are not yet supported and measure '
+                    'argument is "{}" model dependent.'.format(arg_obj.identifier))
+            self._arguments.append(arg_obj)
+
+    @staticmethod
+    def _compress_file(filepath):
+        """Compress file contents to a string."""
+        # TODO: Research better ways to compress the file
+        with open(filepath, 'r') as input_file:
+            content = input_file.read()
+        return content
+
+    @staticmethod
+    def _decompress_to_file(value, filepath):
+        """Write file contents to a file."""
+        with open(filepath, 'w') as output_file:
+            output_file.write(value)
+
+    def ToString(self):
+        return self.__repr__()
+
+    def __repr__(self):
+        return 'Measure: {}'.format(self.identifier)
+
+
+class MeasureArgument(object):
+    """Object representing a single measure argument.
+
+    Args:
+        xml_element: A Python XML Element object taken from the <arguments> section
+            of the measure.xml file.
+
+    Properties:
+        * identifier
+        * display_name
+        * value
+        * default_value
+        * type
+        * type_text
+        * required
+        * description
+        * model_dependent
+        * valid_choices
+    """
+    PYTHON_TYPES = {
+        'Double': float,
+        'Integer': int,
+        'Boolean': bool,
+        'String': str,
+        'Choice': str
+    }
+
+    def __init__(self, xml_element):
+        """Initialize MeasureArgument."""
+        # set up the argument value and default value
+        self._value = None  # will be set by user
+        self._default_value = xml_element.find('default_value').text \
+            if xml_element.find('default_value') is not None else None
+
+        # parse the required properties of the argument
+        self._identifier = xml_element.find('name').text
+        self._type_text = xml_element.find('type').text
+        self._type = self.PYTHON_TYPES[self._type_text]
+        required = xml_element.find('required').text
+        self._required = True if required == 'true' else False
+
+        # parse the optional properties of the argument
+        self._display_name = xml_element.find('display_name').text \
+            if xml_element.find('display_name') is not None else None
```

Comment:
I believe you don't need `else None` for all these case. If it is not set then it will be None.

Reply:
I checked and it is needed. You get a `SyntaxError` otherwise.